### ISSUE TITLE
Random seed limited to 1..10000000

### ISSUE THREAD

**Issue by pxk2k** [cmdr2/stable-diffusion-ui#1035]
**Describe the bug**
When generating a random seed, the limit is set to 10 million when it could be 2**32
The 10M limit is only 0.2% of the seed universe.
 
Thank you for an excellent bundle of easy to install and feature rich software!

**Comment by ollobrains** [comment]
is there any possibiliti of setting this seed number to a full range ?

**Comment by JeLuF** [comment]
You can enter any seed number into the seed field. The limitation only affects randomly generated seeds.

**Comment by pxk2k** [comment]
Thank you for answering my question.

Yes, I understand being able to enter any number in the seed field.

I don't understand why using 10000000 when you could use 2^32 and fix the whole random generator thing.

I am an (old) C guy myself and am curious if this is about any python or code limitation?

Regards, pac
--

**Comment by ollobrains** [comment]
ive already got a pull request up for merging lora models into main models,  so i cant open another pull request, but u should open a pull request, it has to be easy to program in, and if we dont have developers, go and promote the program about say reddit, and other places.

**Comment by JeLuF** [comment]
> I am an (old) C guy myself and am curious if this is about any python or code limitation?

No, it's a readability thing. Longer numbers are harder to remember than shorter numbers.

10 million random numbers will be enough to keep your computer busy for a while, and any seed number is as good as any other one.

**Comment by ollobrains** [comment]
without tredding on any toes, perhaps an option for the user to select 10 million, 100 million or unlimited as an option, the coding change i would image would be pretty straight forward and then leave it up to the user to deicde (with the default for easy use perhaps being 10 million)

**Comment by ollobrains** [comment]
> Why?

why not, what is the downside to implementing it in terms of performance or just accessibility for end users ?

**Comment by pxk2k** [comment]
In my humble opinion, seeds are not remembered but written as metadata and I would rather have software with full functionality than crippled. But it is up to the coders as this is a personal choice and not a technical bug.
Anyway, this is a great piece of software, allowing the easy exploration of a fantastic new technique in image processing. Thank you for the great work.

pac

**Comment by ollobrains** [comment]
agree its a great piece of software, i even link the program direct on my civitai.com model releases to encourage more users to come this way, u might get a few new programmers willing to assist as well i enjoy it that much
